pkg/vfs/utils.go

```diff
@@ -0,0 +1,43 @@
+package vfs
+
+import (
+	"io/fs"
+	"os"
+
+	"encr.dev/pkg/eerror"
+)
+
+func FromDir(workingDir string, predicate func(fileName string) bool) (*VFS, error) {
+	dirFS := os.DirFS(workingDir)
+	rtn := New()
+
+	if predicate == nil {
+		predicate = func(fileName string) bool { return true }
+	}
+
+	err := fs.WalkDir(dirFS, ".", func(path string, info fs.DirEntry, err error) error {
```

Comment:
This doesn't appear to check the `err` value so if something fails it will silently drop files from the resulting FS